src/ps/psreaddata.cpp

```diff
@@ -588,59 +627,105 @@ PetscErrorCode PSReadMatPowerData(PS ps, const char netfile[]) {
+      if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
+        dcline_nblank_lines = dcline_nblank_lines + 1;
+    }
+
+    if (gencost_start_line != -1 && gencost_end_line == -1) {
+      if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
+        gencost_nblank_lines = gencost_nblank_lines + 1;
+    }
+
+    if (loadcost_start_line != -1 && loadcost_end_line == -1) {
+      if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
+        loadcost_nblank_lines = loadcost_nblank_lines + 1;
+    }
+
+    if (genfuel_start_line != -1 && genfuel_end_line == -1) {
+      if (strcmp(line, "\n") == 0 || strcmp(line, "\r\n") == 0)
+        genfuel_nblank_lines = genfuel_nblank_lines + 1;
     }
 
     /* Count the number of pq loads */
     if (bus_start_line != -1 && line_counter >= bus_start_line &&
         bus_end_line == -1) {
       sscanf(line, "%d %d %lf %lf", &extbusnum, &bustype_i, &Pd, &Qd);
+      if (bustype_i == ISOLATED_BUS) { /* Isolated bus */
+        /* put it in the isolated_buses array for later work */
+        ps->isolated_buses[ps->nisolated_buses++] = extbusnum;
+        if (ps->nisolated_buses == buff_num * 10) {
+          /* Reached buffer max, need to allocate more size */
+          buff_num++;
+          ierr = PetscCalloc1(buff_num * 10, &temp_buff);
+          CHKERRQ(ierr);
+          /* Copy over isolated_buses to temp_buff */
+          ierr = PetscMemcpy(temp_buff, ps->isolated_buses,
+                             (buff_num - 1) * 10 * sizeof(PetscInt));
+          ierr = PetscFree(ps->isolated_buses);
+          CHKERRQ(ierr);
+          /* Updated pointer to isolated_buses */
+          ps->isolated_buses = temp_buff;
+        }
+      }
+
       if (!((Pd == 0.0) && (Qd == 0.0)))
-        ps->Nload++;
+        load_numlines++;
       if (extbusnum > ps->maxbusnum)
         ps->maxbusnum = extbusnum;
     }
+
+    /* Count the number of cost-coefficients */
+    if (gencost_start_line != -1 && line_counter >= gencost_start_line &&
+        gencost_end_line == -1) {
+      PetscInt gencost_model, num_cost_coeffs;
+      PetscScalar cost_startup, cost_shutdown;
+      sscanf(line, "%d %lf %lf %d", &gencost_model, &cost_startup,
+             &cost_shutdown, &num_cost_coeffs);
+      if (num_cost_coeffs > num_max_cost_coeffs)
+        num_max_cost_coeffs = num_cost_coeffs;
+    }
+
     line_counter++;
   }
   fclose(fp);
 
-  ps->Nbus = ps->nbus = bus_end_line - bus_start_line - bus_nblank_lines;
-  ps->Ngen = ps->ngen = gen_end_line - gen_start_line - gen_nblank_lines;
-  ps->Nline = ps->nline = br_end_line - br_start_line - br_nblank_lines;
-  ps->nload = ps->Nload;
+  bus_numlines = bus_end_line - bus_start_line - bus_nblank_lines;
+  gen_numlines = gen_end_line - gen_start_line - gen_nblank_lines;
+  br_numlines = br_end_line - br_start_line - br_nblank_lines;
+  dcline_numlines = dcline_end_line - dcline_start_line - dcline_nblank_lines;
```

Comment:
As far as I can tell these variables are `PetscCalloc1` allocated, but never used outside of the assignment operator here?

Reply:
These are the sizes of arrays allocated for bus, gen, branch, and dcline.

Reply:
You are right. They are not anywhere else except for allocating the arrays.